(FM-8217) Port module to litmus

michaeltlombardi opened the pull request:
This commit updates the acceptance testing of the module
to use litmus instead of beaker.

michaeltlombardi commented:
I believe this will fail until puppetlabs/puppet_litmus#147 is merged.

tphoney commented on distelli-manifest.yml:
```diff
@@ -0,0 +1,25 @@
+team-modules/puppetlabs-package:
```
need to change the title

tphoney commented:
the tests seem quick, for travis we could combine the el 6+7 test groups

michaeltlombardi commented:
Acceptance tests are passing in both appveyor and travis, travis failures seem to be gem related.

This PR requires some rebasing prior to merge.

tphoney commented on Gemfile:
```diff
@@ -17,6 +17,7 @@ ruby_version_segments = Gem::Version.new(RUBY_VERSION.dup).segments
 minor_version = ruby_version_segments[0..1].join('.')
 
 group :development do
+  gem 'serverspec'
```
not needed

tphoney replied:
not needed, in puppet-module-gems

tphoney commented on Gemfile:
```diff
@@ -28,11 +29,8 @@ group :development do
-  gem "puppet-module-win-system-r#{minor_version}",   require: false, platforms: [:mswin, :mingw, :x64_mingw]
-  gem "bolt", '~> 1.3',                               require: false if ENV['GEM_BOLT']
+  # gem "puppet_litmus",                                           require: false, platforms: [:ruby, :mswin, :mingw, :x64_mingw] if ENV['PUPPET_GEM_VERSION'].nil? or ENV['PUPPET_GEM_VERSION'] !~ %r{ 5}
+  gem "puppet_litmus", git: 'https://github.com/michaeltlombardi/puppet_litmus', branch: 'bugfix-localhost-run_bolt_task'
```
remove before merge (obviously)

tphoney commented on .sync.yml:
```diff
@@ -3,51 +3,27 @@
   required:
```
you should run a pdksync as well, to make sure that the next pdksync will not destroy any changes you have put in

michaeltlombardi commented:
Updated per comments and added a pdksync. I think the appveyor tests will continue to fail until a new version of the litmus gem is released containing the changes from puppetlabs/puppet_litmus#147.

tphoney commented on provision.yaml:
```diff
@@ -0,0 +1,19 @@
+waffle_deb:
+  provisioner: docker_exp
+  images: ['waffleimage/debian8', 'waffleimage/debian9', 'waffleimage/ubuntu14.04', 'waffleimage/ubuntu16.04', 'waffleimage/ubuntu18.04']
+waffle_el6:
```
seeing as the tests are quick to run, we should combine the el6 and el7 groups. And just run one group, not 2 for EL

michaeltlombardi replied:
@tphoney Looking at the Travis config, it looks like it is distinguishing between the EL5 and EL6 platforms? If those can be safely squashed I'm happy to do this, digging in now.

michaeltlombardi commented:
@tphoney yeah I'm gonna add a litmus release to my to-do list for today, at least to prep it all.

pmcmaw commented:
Resolves #82
